From: Nicolas Michaut <nicolas@yumilab.com>
Subject: [PATCH] video: center the built-in boot logo
Date: 2026-07-27

The built-in logo (CONFIG_VIDEO_LOGO) is hardcoded to the top-right
corner of the screen. Draw it centered instead: video_bmp_display()
already knows how to center a bitmap when passed BMP_ALIGN_CENTER.

Signed-off-by: Nicolas Michaut <nicolas@yumilab.com>
---
 drivers/video/video-uclass.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/video/video-uclass.c
+++ b/drivers/video/video-uclass.c
@@ -14,6 +14,7 @@
 #include <malloc.h>
 #include <mapmem.h>
 #include <spl.h>
+#include <splash.h>
 #include <stdio_dev.h>
 #include <video.h>
 #include <video_console.h>
@@ -597,7 +598,8 @@
 	u8 *data = SPLASH_START(u_boot_logo);
 	int ret;
 
-	ret = video_bmp_display(dev, map_to_sysmem(data), -4, 4, true);
+	ret = video_bmp_display(dev, map_to_sysmem(data), BMP_ALIGN_CENTER,
+				BMP_ALIGN_CENTER, true);
 
 	return 0;
 }
-- 
2.43.0
